src/main/java/com/blackducksoftware/integration/hub/alert/datasource/relation/manager/HubUserManager.java

```diff
@@ -0,0 +1,240 @@
+import com.blackducksoftware.integration.hub.alert.web.model.HubUsersConfigWrapper;
+import com.blackducksoftware.integration.hub.alert.web.model.ProjectVersionConfigWrapper;
+
+@Component
+public class HubUserManager {
+    private final Logger logger = LoggerFactory.getLogger(HubUsersConfigWrapperActions.class);
+
+    private final HubUsersRepository hubUsersRepository;
+    private final HubUserFrequenciesRepository hubUserFrequenciesRepository;
+    private final HubUserEmailRepository hubUserEmailRepository;
+    private final HubUserHipChatRepository hubUserHipChatRepository;
+    private final HubUserSlackRepository hubUserSlackRepository;
+    private final HubUserProjectVersionsRepository hubUserProjectVersionsRepository;
+    private final ObjectTransformer objectTransformer;
+
+    @Autowired
+    public HubUserManager(final HubUsersRepository hubUsersRepository, final HubUserFrequenciesRepository hubUserFrequenciesRepository, final HubUserEmailRepository hubUserEmailRepository,
+            final HubUserHipChatRepository hubUserHipChatRepository, final HubUserSlackRepository hubUserSlackRepository, final HubUserProjectVersionsRepository hubUserProjectVersionsRepository, final ObjectTransformer objectTransformer) {
+        this.hubUsersRepository = hubUsersRepository;
+        this.hubUserFrequenciesRepository = hubUserFrequenciesRepository;
+        this.hubUserEmailRepository = hubUserEmailRepository;
+        this.hubUserHipChatRepository = hubUserHipChatRepository;
+        this.hubUserSlackRepository = hubUserSlackRepository;
+        this.hubUserProjectVersionsRepository = hubUserProjectVersionsRepository;
+        this.objectTransformer = objectTransformer;
+    }
+
+    public Long saveConfig(final HubUsersConfigWrapper wrapper) throws AlertException {
+        HubUsersEntity hubUsersEntity;
+        Long configId = objectTransformer.stringToLong(wrapper.getId());
+        if (configId == null || !hubUsersRepository.exists(configId)) {
+            hubUsersEntity = hubUsersRepository.save(new HubUsersEntity(wrapper.getUsername()));
+        } else {
+            hubUsersEntity = hubUsersRepository.findOne(configId);
+        }
+        configId = hubUsersEntity.getId();
+        saveFrequency(configId, wrapper.getFrequency());
+        saveChannelId(configId, wrapper.getEmailConfigId(), hubUserEmailRepository, HubUserEmailRelation.class);
+        saveChannelId(configId, wrapper.getHipChatConfigId(), hubUserHipChatRepository, HubUserHipChatRelation.class);
+        saveChannelId(configId, wrapper.getSlackConfigId(), hubUserSlackRepository, HubUserSlackRelation.class);
+        saveProjectVersions(configId, wrapper.getProjectVersions());
+
+        return configId;
+    }
+
+    public void deleteConfig(final String id) {
+        deleteConfig(objectTransformer.stringToLong(id));
+    }
+
+    public void deleteConfig(final Long id) {
+        if (doesConfigExist(id)) {
+            hubUsersRepository.delete(id);
```

Comment:
Perform the deletion in the opposite order of the save.  So the deletion of the user object should be the last thing that is deleted.